### PR TITLE
api: add PodMigrationJob API

### PR DESCRIPTION
**Pull request by eahydra**
Signed-off-by: Joseph <joseph.t.lee@outlook.com>

### Ⅰ. Describe what this PR does

add PodMigrationJob API defined in proposal #358 

### V. Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `make test`

### CODE REVIEW

File: apis/scheduling/v1alpha1/pod_migration_job_types.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	TimeoutInSeconds *int32 `json:"timeoutInSeconds,omitempty"`
+
+	// Mode represents the operating mode of the Job
+	// Default is PodMigrationJobModeReservationFirst
+	// +optional
+	Mode PodMigrationJobMode `json:"mode,omitempty"`
+
+	// PodRef represents the Pod that be migrated
+	// +required
+	PodRef *corev1.ObjectReference `json:"podRef,omitempty"`
```
Comment (jasonliu747) [inline apis/scheduling/v1alpha1/pod_migration_job_types.go:44]: to be required, or not to be. that is a question.

Comment (eahydra) [reply apis/scheduling/v1alpha1/pod_migration_job_types.go]: fixed

**Review by jasonliu747 — COMMENTED** [review]
/lgtm

**Pull request merged**